storm/src/main/storm/mesos/MesosSupervisor.java

```diff
@@ -56,17 +55,22 @@
   Map _conf;
   // Store state on port assignments arriving from MesosNimbus as task-launching requests.
   private static final TaskAssignments _taskAssignments = TaskAssignments.getInstance();
-  // What is the storm-core supervisor's view of the assigned ports?
-  AtomicReference<Set<Integer>> _supervisorViewOfAssignedPorts = new AtomicReference<Set<Integer>>(new HashSet<Integer>());
 
   public static void main(String[] args) {
-    org.apache.storm.daemon.supervisor.launch(new MesosSupervisor());
+    Map<String, Object> conf = ConfigUtils.readStormConfig();
+
+    try {
+      Supervisor supervisor = new Supervisor(conf, null, new MesosSupervisor());
+      supervisor.launch();
```

Comment:
There's another potential issue I just noticed.  The following code is only invoked within the `launchDaemon()` method, and not from `launch()` method:
* https://github.com/apache/storm/blob/v1.0.3/storm-core/src/jvm/org/apache/storm/daemon/supervisor/Supervisor.java#L244-L251

So can you please try using `supervisor.launchDaemon()` instead of just `supervisor.launch()`?